client/pkg/transport/tls_test.go

```diff
@@ -0,0 +1,36 @@
+package transport
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestValidateSecureEndpoints(t *testing.T) {
+	tlsInfo, err := createSelfCert(t)
+	if err != nil {
+		t.Fatalf("unable to create cert: %v", err)
+	}
+
+	remoteAddr := func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(r.RemoteAddr))
+	}
+	srv := httptest.NewServer(http.HandlerFunc(remoteAddr))
+	defer srv.Close()
+
+	insecureEps := []string{
+		"http://" + srv.Listener.Addr().String(),
+		"invalid remote address",
+	}
+	if _, err := ValidateSecureEndpoints(*tlsInfo, insecureEps); err == nil || !strings.Contains(err.Error(), "is insecure") {
+		t.Error("validate secure endpoints should fail")
+	}
+
+	secureEps := []string{
```

Comment:
Looks like you sequentially test different cases, fist insecure later secure, also you don't  have you considered using subtests?

Reply:
OK, I will using subtests.